src/coreclr/jit/lsra.cpp

```diff
@@ -5874,6 +5874,8 @@ void LinearScan::allocateRegisters()
             continue;
         }
 
+        assert(!currentRefPosition.isPhysRegRef);
```

Comment:
Also added this to match `allocateRegistersMinimal` (forgot it in the previous PR)